xds/src/test/java/io/grpc/xds/XdsLoadBalancerTest.java

```diff
@@ -149,121 +307,306 @@ public void canHandleEmptyAddressListFromNameResolution() {
+        + "\"balancerName\" : \"dns:///balancer.example.com:8443\","
+        + "\"childPolicy\" : [{\"supported_1\" : {\"key\" : \"val\"}}, {\"unsupported_2\" : {}}],"
+        + "\"fallbackPolicy\" : [{\"unsupported\" : {}}, {\"supported_1\" : {\"key\" : \"val\"}}]"
+        + "}}";
+    @SuppressWarnings("unchecked")
+    Map<String, Object> lbConfig2 = (Map<String, Object>) JsonParser.parse(lbConfigRaw);
+    attrs = Attributes.newBuilder().set(ATTR_LOAD_BALANCING_CONFIG, lbConfig2).build();
 
     lb.handleResolvedAddressGroups(Collections.<EquivalentAddressGroup>emptyList(), attrs);
 
-    assertThat(lb.getXdsLbState().childPolicy).isNotNull();
+    assertThat(lb.getXdsLbStateForTest().childPolicy).isNotNull();
+
+    // verify oobChannel is unchanged
+    verify(helper, times(2)).createOobChannel(Matchers.<EquivalentAddressGroup>any(), anyString());
+    verify(oobChannel1)
+        .newCall(Matchers.<MethodDescriptor<?, ?>>any(), Matchers.<CallOptions>any());
+    verify(oobChannel2)
+        .newCall(Matchers.<MethodDescriptor<?, ?>>any(), Matchers.<CallOptions>any());
+    verifyNoMoreInteractions(oobChannel3);
+  }
+
+  @Test
+  public void fallback_AdsNotWorkingYetTimerExpired() throws Exception {
+    lb.handleResolvedAddressGroups(
+        Collections.<EquivalentAddressGroup>emptyList(), standardModeWithFallback1Attributes());
+
+    fakeClock.forwardTime(10, TimeUnit.SECONDS);
+
+    ArgumentCaptor<Attributes> captor = ArgumentCaptor.forClass(Attributes.class);
+    verify(fakeBalancer1).handleResolvedAddressGroups(
+        Matchers.<List<EquivalentAddressGroup>>any(), captor.capture());
+    assertThat(captor.getValue().get(ATTR_LOAD_BALANCING_CONFIG))
+        .containsExactly("supported_1", new HashMap<String, Object>());
+  }
+
+  @Test
+  public void fallback_AdsWorkingTimerExpired() throws Exception {
+    lb.handleResolvedAddressGroups(
+        Collections.<EquivalentAddressGroup>emptyList(), standardModeWithFallback1Attributes());
+
+    serverResponseWriter.onNext(DiscoveryResponse.getDefaultInstance());
+
+    assertThat(fakeClock.getPendingTasks()).isEmpty();
+    verify(fakeBalancer1, never()).handleResolvedAddressGroups(
+        Matchers.<List<EquivalentAddressGroup>>any(), Matchers.<Attributes>any());
+  }
+
+  @Test
+  public void fallback_AdsErrorAndNoActiveSubchannel() throws Exception {
+    lb.handleResolvedAddressGroups(
+        Collections.<EquivalentAddressGroup>emptyList(), standardModeWithFallback1Attributes());
+
+    serverResponseWriter.onError(new Exception("fake error"));
+
+    ArgumentCaptor<Attributes> captor = ArgumentCaptor.forClass(Attributes.class);
+    verify(fakeBalancer1).handleResolvedAddressGroups(
+        Matchers.<List<EquivalentAddressGroup>>any(), captor.capture());
+    assertThat(captor.getValue().get(ATTR_LOAD_BALANCING_CONFIG))
+        .containsExactly("supported_1", new HashMap<String, Object>());
+
+    fakeClock.forwardTime(10, TimeUnit.SECONDS);
```

Comment:
Maybe 
```java
assertThat(fakeClock.forwardTime(10, TimeUnit.SECONDS)).isEqualTo(1);
assertThat(fakeClock.getPendingTasks()).isEmpty();
```
to verify that the timer did expire, but had no effect.

Reply:
Done.